src/lib.rs

```diff
@@ -64,6 +67,20 @@ pub fn run(config: &Config) -> Result<(), Box<dyn Error>> {
             crate::cli::print_win()
         }
     }
+}
+
+pub fn run(config: &Config) -> Result<(), Box<dyn Error>> {
+    // run galgo with potential rematches
+    let mut rematch = true;
+
+    while rematch == true {
+        run_game(config);
+
+        let rematch_input = crate::cli::take_user_input("Do you want a rematch? (yes/no)");
+        if rematch_input.contains("n") {
+            rematch = false
+        }
+    }
```

Comment:
[nit]: What do you think about taking the user's input on whether to start a game in the loop condition in order to get rid of the `rematch` state handling?

```suggestion

    while crate::cli::take_user_input("Start a new game? (yes/no)")
        .eq_ignore_ascii_case("yes") {
            run_game(config);
    }
```